src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -124,64 +125,74 @@ def acr_show(registry_name, resource_group_name=None):
-               resource_group_name=None,
-               storage_account_name=None,
-               tags=None,
-               enable_admin=False,
-               disable_admin=False):
-    '''Update a container registry.
-    :param str registry_name: The name of container registry
-    :param str resource_group_name: The name of resource group
-    :param str storage_account_name: The name of storage account
-    :param dict tags: The set of tags
-    :param bool enable_admin: Enable admin user
-    :param bool disable_admin: Disable admin user
-    '''
+def acr_update_set(registry_name,
+                   resource_group_name=None,
+                   parameters=None):
     registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
 
-    # Set admin_user_enabled
-    admin_user_enabled = None
-    if disable_admin:
-        admin_user_enabled = False
-    if enable_admin:
-        admin_user_enabled = True
-
-    # Set tags
-    newTags = registry.tags #pylint: disable=no-member
-    if isinstance(tags, dict):
-        if tags:
-            for key in tags:
-                if tags[key]:
-                    newTags[key] = tags[key]
-                elif key in newTags:
-                    del newTags[key]
-        else:
-            newTags = {}
+    admin_user_enabled = parameters.admin_user_enabled if parameters.admin_user_enabled else False
 
     # Set storage account
     storage_account = None
-    if storage_account_name:
-        storage_account_key = get_access_key_by_storage_account_name(storage_account_name)
+    if parameters.storage_account_name:
+        storage_account_key = get_access_key_by_storage_account_name(parameters.storage_account_name)
         storage_account = StorageAccountProperties(
-            storage_account_name,
+            parameters.storage_account_name,
             storage_account_key
         )
 
+    # Set tags
+    tags = parameters.tags #pylint: disable=no-member
+    if not isinstance(tags, dict):
+        tags = {}
+
     client = get_acr_service_client().registries
 
     return client.update(
         resource_group_name, registry_name,
         RegistryUpdateParameters(
-            tags=newTags,
-            admin_user_enabled=admin_user_enabled,
-            storage_account=storage_account if storage_account else None
+            tags=tags,
+            storage_account=storage_account,
+            admin_user_enabled=admin_user_enabled
         )
     )
 
+def acr_update_get(registry_name,
+                   resource_group_name=None):
+    registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
+
+    tags = registry.tags
+    admin_user_enabled = registry.admin_user_enabled
+    storage_account_name = registry.storage_account.name
+
+    return ACRUpdateParameters(
+        tags=tags,
+        admin_user_enabled=admin_user_enabled,
+        storage_account_name=storage_account_name
+    )
+
+class ACRUpdateParameters:
```

Comment:
You don't need this class. You can just have the update parameters in your custom method signature like you did. before.